src/arch/riscv/traps.S

```diff
@@ -135,6 +135,12 @@ handle_syscall:
 
 /* Not an interrupt or a syscall */
 exception:
+  /* check if SPP (sstatus[8]) was 1, i.e. exception from the kernel */
+  li    t1, 0x100
```

Comment:
It there a real gain doing this check in the asm code? I could also happen in the C code of `c_handle_exception() then. Rational is, that if we are here, we have pushed all registers on the stack anyway without crashing again, and `c_handle_kernel_exception()` tries to print something before calling `halt()`. So in the worst case all this gets stuck in a loop anyway and we never reach `halt()`. Otherwise, the machine still seems to work somehow, so we can also do this in C code exception hander

Reply:
If it's under an ifdef for debug builds, verification does not see it.